feat(cpu): detect hypervisor on x86_64 and arm64 via CPUID

whatever125 opened the pull request:
Added detection of the virtualization platform on x86/arm64 archs using klauspost/cpuid lib.

- Preserved existing s390x implementation using `/proc/sysinfo`.
- On x86 and arm64 `cpu.model.hypervisor` now reports hypervisor type, or `none` for bare-metal, and raw CPUID string for unknown hypervisors.

Resolves #2267

marquiz commented on source/cpu/cpu.go:
```diff
@@ -274,33 +274,78 @@ func getCPUModel() map[string]string {
+	if cpuid.CPU.VM() {
+		hv := strings.ToLower(cpuid.CPU.HypervisorVendorID.String())
+	
+		switch hv {
```
```suggestion
		switch hv := strings.ToLower(cpuid.CPU.HypervisorVendorID.String()); hv {
```

marquiz commented on source/cpu/cpu.go:
```diff
@@ -274,33 +274,78 @@ func getCPUModel() map[string]string {
+		case "acrn":
+			return "acrn", nil
+		case "sre":
+			return "sre", nil
```
Couldn't we put all these cases where we don't mangle the value under the default case?


```suggestion
		default:
			return hv, nil
```

whatever125 commented:
Thank you for the feedback @marquiz. I've updated the code as you suggested:
1. Split into helper functions, updated fallback logic
2. Used default in switch case. Now only `cpuid.VendorUnknown` is treated separately to get vendor from `cpuid.CPU.HypervisorVendorString`. Otherwise, the default `cpuid.CPU.HypervisorVendorID.String()` is used
3. Got rid of string handling for simplicity and maintainability

marquiz commented on source/cpu/cpu.go:
```diff
@@ -274,33 +274,65 @@ func getCPUModel() map[string]string {
+		hv, discovered := getHypervisorFromProcSysinfo()
+		if discovered {
+			return hv, nil
+		}
```
If this does not find anything (`discovered == false`), do we want to fall back to `getHypervisorFromCPUID()`? Probably not, or WDYT?
